### PR TITLE
use synchronous ewasm-kernel

### PR DESCRIPTION
**Pull request by cdetrio**
transpiler branch that uses the synchronous ewasm methods (i.e. no wasm callbacks).

### CODE REVIEW

File: wasm/callback_128.wast [hunk 1]
```diff
@@ -1,4 +1,5 @@
 (func $callback_128
+  (export "2")
```
Comment (axic) [inline wasm/callback_128.wast]: I think these exports should be added in the JS part with `(export "2" $callback_128)` so there would be a single place of truth for the indexes.

File: index.js [hunk 2]
```diff
@@ -480,9 +508,23 @@ exports.buildModule = function (funcs, imports = [], exports = [], cbs = []) {
   for (let func of funcs) {
     funcStr += func
   }
+  // exports is always empty, not sure what the intended use was?
   for (let exprt of exports) {
     funcStr += `(export "${exprt}" (func $${exprt}))`
```
Comment (axic) [inline index.js]: `main` is exported

File: wasm/callback_128.wast (hunk 1 above)
Comment (axic) [reply wasm/callback_128.wast]: Actually you could make these part of the `exports` list you had a comment about below :)

File: index.js [hunk 3]
```diff
@@ -288,14 +299,29 @@ exports.evm2wast = function (evmCode, opts = {
         i = findNextJumpDest(evmCode, i)
         break
       default:
-        if (callbackFuncs.has(op.name)) {
+        if (USE_ASYNC_API && callbackFuncs.has(op.name)) {
```
Comment (axic) [inline index.js]: Should make `USE_ASYNC_API` part of `opts.useAsyncAPI` so we can build both from the commandline.

**Comment by axic** [comment]
@cdetrio is this still needed or is this an obsolete PR?

**Comment by cdetrio** [comment]
obsoleted by https://github.com/ewasm/evm2wasm/pull/180

**Pull request closed without merging**